multicore_runtime/wsgi_config.py

```diff
@@ -0,0 +1,216 @@
+  If the user chooses to define the `webapp_add_wsgi_middleware` function
+  there, then we will use it to wrap app scripts. This is used for e.g.
+  appstats. Many user apps do not have or need env_config, so failure
+  to import it or failure to find `webapp_add_wsgi_middleware` is not an
+  error.
+
+  Returns:
+    The appengine_config.webapp_add_wsgi_middleware function or None.
+  """
+  try:
+    import appengine_config  # pylint: disable=g-import-not-at-top
+    return getattr(appengine_config, 'webapp_add_wsgi_middleware', None)
+  except ImportError:
+    return None
+
+
+def static_app_for_handler(handler):
+  """Returns a WSGI app that serves static files as directed by the handler.
+
+  Args:
+    handler: An individual handler from appinfo_external.handlers
+      (appinfo.URLMap)
+
+  Returns:
+    A static file-serving WSGI app closed over the handler information.
+  """
+  url_re = handler.url  # handler.url is a regex, despite the name.
+  files = handler.static_files
+  upload = handler.upload
+  if not files:
+    if handler.static_dir:
+      # If static_files is not set, convert static_dir to static_files and also
+      # modify the url regex accordingly. See the appinfo.URLMap docstring for
+      # more information.
+      url_re = static_dir_url(handler)
+      files = '{dir}{backref}'.format(dir=handler.static_dir, backref=r'/\1')
+      upload = '{dir}{re}'.format(dir=handler.static_dir, re='/.*')
+    else:
+      # Neither static_files nor static_dir is set; log an error and return.
+      logging.error('No script, static_files or static_dir found for %s',
+                    handler)
+      return None
+  return static_app_for_regex_and_files(url_re, files, upload,
+                                        mime_type=handler.mime_type)
+
+
+def static_dir_url(handler):
+  """Converts a static_dir regex into a static_files regex if needed.
+
+  See the appinfo.URLMap docstring for more information.
+
+  Args:
+    handler: A handler (appinfo.URLMap)
+
+  Returns:
+    A modified url regex
+  """
+  if not handler.script and not handler.static_files and handler.static_dir:
+    return '{url}/(.*)'.format(url=handler.url)
+  else:
+    return handler.url
+
+
+def load_user_scripts_into_handlers(handlers):
+  """Preloads user scripts, wrapped in env_config middleware if present.
+
+  Args:
+    handlers: appinfo.handlers data as provided by get_module_config()
+
+  Returns:
+    A list of tuples suitable for configuring the dispatcher() app,
+    where the tuples are (url, app):
+      - url_re: The url regular expression which matches this handler.
+      - app: The fully loaded app corresponding to the script.
+  """
+  # `if x.login == appinfo.LOGIN_OPTIONAL` disables loading handlers
+  # that require login or admin status entirely. This is a temporary
+  # measure until handling of login-required handlers is implemented
+  # securely.
+  loaded_handlers = []
+  for x in handlers:
```

Comment:
Actually, part of rolling this out is that now you can use a good variable name, like `handler`, instead of a cryptic one, such as `x` :-)